src/main/java/io/lettuce/core/resource/Transports.java

```diff
@@ -74,7 +74,7 @@ public static Class<? extends DatagramChannel> datagramChannelClass() {
     public static class NativeTransports {
 
         static EventLoopResources RESOURCES = KqueueProvider.isAvailable() ? KqueueProvider.getResources()
-                : IOUringProvider.isAvailable() ? IOUringProvider.getResources() : EpollProvider.getResources();
+                : EpollProvider.isAvailable() ? EpollProvider.getResources() : IOUringProvider.getResources();
```

Comment:
I would leave the order as is and only apply a tweak in the domain socket class retrieval. I'll take care of this during the merge.